tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -470,32 +514,35 @@ void ROOT::Experimental::RNTupleJoinProcessor::SetModel(std::unique_ptr<ROOT::RN
          auto &joinDefaultEntry = fModel->GetDefaultEntry();
          for (const auto &fieldName : auxModels[i]->GetFieldNames()) {
             auto valuePtr = auxDefaultEntry.GetPtr<void>(fieldName);
-            joinDefaultEntry.BindValue(fNTuples[i + 1].fNTupleName + "." + fieldName, valuePtr);
+            joinDefaultEntry.BindValue(fAuxiliaryProcessors[i]->GetProcessorName() + "." + fieldName, valuePtr);
          }
       }
    }
 
    fModel->Freeze();
 }
 
-void ROOT::Experimental::RNTupleJoinProcessor::ConnectFields()
+void ROOT::Experimental::RNTupleJoinProcessor::SetEntryPointers(const ROOT::REntry &entry,
+                                                                std::string_view fieldNamePrefix)
 {
-   for (auto &[_, fieldContext] : fFieldContexts) {
-      Internal::RPageSource &pageSource =
-         fieldContext.IsAuxiliary() ? *fAuxiliaryPageSources.at(fieldContext.fNTupleIdx - 1) : *fPageSource;
-      ConnectField(fieldContext, pageSource, *fEntry);
+   for (const auto &value : *fEntry) {
+      std::string fieldName = value.GetField().GetQualifiedFieldName();
+      auto valuePtr = fieldNamePrefix.empty() ? entry.GetPtr<void>(fieldName)
+                                              : entry.GetPtr<void>(std::string(fieldNamePrefix) + "." + fieldName);
+
+      fEntry->BindValue(fieldName, valuePtr);
    }
-}
 
-void ROOT::Experimental::RNTupleJoinProcessor::SetEntryPointers(const ROOT::REntry &entry)
-{
-   for (const auto &[_, fieldContext] : fFieldContexts) {
-      auto fieldName = fieldContext.GetProtoField().GetQualifiedFieldName();
-      if (fieldContext.IsAuxiliary()) {
-         fieldName = fNTuples[fieldContext.fNTupleIdx].fNTupleName + "." + fieldName;
+   fPrimaryProcessor->SetEntryPointers(*fEntry);
```

Comment:
Right now this is not passing on `fieldNamePrefix` - I assume that's the problem with supporting joins of joins? If so, maybe add a `FIXME` comment?